### ISSUE TITLE
[Bug]: Type of `input_arg` seems incorrect

### ISSUE THREAD

**Issue by mattgodbolt** [orchidjs/tom-select#648]
### Bug description

When creating a TomSelect the `input_arg` is `string|TomInput`. `TomInupt` extneds `HTMLElement`, but that means any type cheker gets angry with a regular `HTMLElement` For example I get:

```
[tsl] ERROR in /home/matthew/dev/ce/compiler-explorer/static/panes/cfg-view.ts(172,47)
      TS2345: Argument of type 'HTMLElement' is not assignable to parameter of type 'string | TomInput'.
```

### Expected behavior

I'm not sure - maybe I'm using this wrong but I don't know whether it's OK to ignore this error.

### Steps to reproduce

Sorry - couldn't get this to happen injsfiddle (I'm not certain how to get tit to do type checking)

### Additional context

Typescript 4.9.5.node v20

**Comment by oyejorge** [comment]
Is there a strong argument for using TomSelect on elements other than input
and select?

The TomInput type
<https://github.com/orchidjs/tom-select/blob/master/src/types/core.ts#L6>
is defined as an HTMLElement with attributes: value, disabled,
validity..etc. It was done this way because the closest common ancestor for
HTMLInputElement and HTMLSelectElement is the HTMLElement type. However,
since HTMLElement doesn't have the value, disabled, or validity attributes
that TomSelect uses, you'll see this error.

**Comment by mattgodbolt** [comment]
Hi @oyejorge - I don't know? I suspect not. In our code we assert that it's an HTMLSelectElement but that's not a `TomInput` in Typescript types. I don't know enough about the intention of TomSelect here compared to the actual types I get from the DOM. I just know that this changed when updating! I'm having a devil of a time updateing Tom-Select from 2.0.3 as we get so many type errors, and it's also the library holind us back from Typescript 5 on our side, so I don't know if I'm doing something wrong, or if I need to accept it and cast the heck out of everything to get the compiler happy!

**Comment by mattgodbolt** [comment]
We:
```
171         assert(selector instanceof HTMLSelectElement, '.function-selector is not an HTMLSelectElement');
```

directly before we cal TomSelect. How is it supposed to be called if none of the DOM types actually match the TomInput type? Do you have any tips for what I'm doing wrong? :) thanks!

**Comment by oyejorge** [comment]
Are you saying this error 
> Argument of type 'HTMLElement' is not assignable to parameter of type 'string | TomInput'.

comes after
> assert(selector instanceof HTMLSelectElement, '.function-selector is not an HTMLSelectElement');

**Comment by oyejorge** [comment]
I just cloned a fresh copy of compiler-explorer, updated tom-select to 2.3.1 and ran ```npm run ts-check``` and didn't see any errors.

**Comment by mattgodbolt** [comment]
@oyejorge yes, that error happens after the assert.

I'll try updating to 2.3.1 and if I can get it to repro with that I'll publish full repro steps, else I'll gladly close this! Thank you!

**Comment by mattgodbolt** [comment]
Well: I'm beyond glad to report this is fixed with 2.3.1 and I can't see how it was ever a problem! Thanks @oyejorge for your patience on this! :blush:
